README.md

```diff
@@ -852,6 +852,31 @@ process. To get around this, increment the current value of
 `AZUL_DEPLOYMENT_INCARNATION` in the deployment's `environment.py` file, then
 redeploy.
 
+## Unexpected warnings cause tests to fail in `tearDownClass`
+
+If a failure occurs due to an unexpected warning in a `tearDownClass`, add the 
+warning to the list of permitted warnings found in
+[`AzulTestCase`](test/azul_test_case.py) and commit the modifications. 
+
+Due to how unit tests are discovered it's possible that a warning is emitted
+when a module is loaded outside the scope of the `AzulTestCase` context manager.
+The warnings that are emitted and caught are dependent on the order in which 
+modules are loaded. For example, if there was an unpermitted warning emitted in a 
+test case located at `test/service/test_app_logging.TestServiceAppLogging`. It's 
+possible that running 
+`(cd test && python -m unittest service.test_app_logging.TestServiceAppLogging)`
+will not emit the warning within the scope of the `AzulTestCase` context 
+manager, but instead emit the warning when the tests are loaded.
+When running a test case from the directory that contains it, 
+`(cd test/service && python -m unittest test_app_logging.TestServiceAppLogging)`
+the context manager might catch the warning and fail appropriately. When running
+the test from a different directory, the test loader for some reason traverses 
```

Comment:
"for some reason" could also maybe use more investigation, but if Hannes doesn't mind, it's fine with me :shrug:.